build_tools/cmake/iree_benchmark_suite.cmake

```diff
@@ -67,7 +69,7 @@ function(iree_benchmark_suite)
   cmake_parse_arguments(
     PARSE_ARGV 0
     _RULE
-    ""
+    "COLLECT_COMPILATION_STATISTICS"
     "DRIVER;TARGET_BACKEND;TARGET_ARCHITECTURE"
     "BENCHMARK_MODES;BENCHMARK_TOOL;MODULES;TRANSLATION_FLAGS;RUNTIME_FLAGS"
```

Comment:
Could we decouple compilation and compilation statistics collection from execution time benchmarking? It looks like this PR offers no way to track offline statistics for a configuration without tightly coupling that same configuration to an execution time benchmark.

(I have a feeling the answer will be "maybe in the future", but these sorts of infrastructure configurations are harder to split apart once they settle)